src/chebcoefs.jl

```diff
@@ -0,0 +1,43 @@
+"""
+   plan_fft(vals::AbstractArray{<:Number,N})
+
+Create an FFTW real-to-real (R2R) transformation plan for the first axis of a given multidimensional array `vals`. 
+In practise, the `vals` array is the power spectrum P(k,χ). The first axis should then contain the wavenumbers `k`.
+
+# Arguments
+- `vals::AbstractArray{<:Number, N}`: The input array of any numerical type with `N` dimensions.
+
+# Returns
+- `p::FFTW.rFFTWPlan`: An FFTW plan object for transforming `vals` with the appropriate R2R transformations. This plan can be applied using the `*` operator (e.g., `transformed_vals = p * vals`).
+
+"""
+function plan_fft(vals::AbstractArray{<:Number,N}) where {N}
+    kind = map(n -> n > 1 ? FFTW.REDFT00 : FFTW.DHT, size(vals)[1])
+    p = FFTW.plan_r2r(deepcopy(vals), kind, [1]; flags=FFTW.PATIENT, timelimit=Inf)   
+                                                                                    
+    return p 
+end
```

Comment:
We will focus on this in a later stage, but we should **specialize** the code here.
In particular, we know that the input array has two axes.
For now, this is fine.